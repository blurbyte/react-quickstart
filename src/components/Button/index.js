import styled from 'styled-components';

const Button = styled.button`
  background: rgba(40,40,40,0.05);
  border: none;
  font-family: 'OpenSans Bold', Arial, Helvetica, sans-serif;
  color: #ff4326;
  font-size: 1.4rem;
  text-transform: uppercase;
  line-height: 3.4rem;
  padding: 0 2rem;
  cursor: pointer;
  border-radius: 6px;
  transition: all 0.4s ease;

  &:hover {
    background: rgba(40,40,40,0.1);
  }

  &:active {
    background: rgba(255,67,38,0.3);
  }

  &:disabled {
    color: rgba(40,40,40, 0.3);
    background: rgba(40,40,40,0.05);
    cursor: default;
  }
`;

export default Button;
