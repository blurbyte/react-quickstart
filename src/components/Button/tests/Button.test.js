import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Button from '../index';

describe('<Button />', () => {
  it('should render correctly', () => {
    // component init
    const component = renderer.create(
      <Button />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
