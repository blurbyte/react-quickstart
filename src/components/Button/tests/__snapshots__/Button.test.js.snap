// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
.iSHzmE {
  background: rgba(40,40,40,0.05);
  border: none;
  font-family: 'OpenSans Bold', Arial, Helvetica, sans-serif;
  color: #ff4326;
  font-size: 1.4rem;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  line-height: 3.4rem;
  padding: 0 4rem;
  cursor: pointer;
  border-radius: 6px;
  -webkit-transition: all 0.4s ease;
  transition: all 0.4s ease;
}

.iSHzmE:hover {
  background: rgba(40,40,40,0.1);
}

.iSHzmE:active {
  background: rgba(255,67,38,0.3);
}

.iSHzmE:disabled {
  color: rgba(40,40,40, 0.3);
  background: rgba(40,40,40,0.05);
  cursor: default;
}

<button
  className="sc-bdVaJa iSHzmE"
/>
`;
